src/artifacts.js

```diff
@@ -423,6 +643,99 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
   this.monitor.reportError(err);
 };
 
+/** Complete artifact */
+api.declare({
+  method:     'put',
+  route:      '/task/:taskId/runs/:runId/artifacts/:name(*)',
+  name:       'completeArtifact',
+  stability:  API.stability.experimental,
+  scopes: [
+    [
+      'queue:create-artifact:<name>',
+      'assume:worker-id:<workerGroup>/<workerId>',
+    ], [
+      'queue:create-artifact:<taskId>/<runId>',
+    ],
+  ],
+  deferAuth:  true,
+  input:      'put-artifact-request.json#',
+  output:     'put-artifact-response.json#',
+  title:      'Complete Artifact',
+  description: 'tbd',
```

Comment:
Add description... outline when this should be called, it's only for `blob` artifacts right?